File: 3. DSA/02. C++/01. Basics/Question13.cpp
// Print nth term of Fibonacci Series

#include <iostream>
using namespace std;

int main()
{
    int n;
    cout << "Enter number: ";
    cin >> n;

    int fValue = 0, sValue = 1, tValue;
    // fValue = firstValue, sValue = secondValue, tValue = thirdValue

    if(n == 1)
    {
        cout << "\nFirst term of Fibonacci Series is " << 0 << endl;
        return 0;
    }

    if(n == 2)
    {
        cout << "\nSecond term of Fibonacci Series is " << 1 << endl;
        return 1;
    }

    for(int i = 3; i <= n; i++)
    {
        tValue = fValue + sValue;
        fValue = sValue;
        sValue = tValue;
    }

    cout << endl << n << "th term of Fibonacci Series is " << tValue << endl;
    return 0;
}

File: 3. DSA/04. Python/01. Basics/Question13.py
# Print nth term of Fibonacci Series

n = int(input("Enter the number: "))
fValue, sValue = 0, 1
tValue = fValue + sValue 

for i in range (1, n - 2):
    fValue, sValue = sValue, tValue
    tValue = fValue + sValue

print(f"{n}th term of Fibonacci Series:", tValue)